src/huggingface_hub/hub_mixin.py

```diff
@@ -208,57 +214,38 @@ def _from_pretrained(
+        token: Optional[str] = None,
+        branch: Optional[str] = None,
         config: Optional[dict] = None,
     ) -> str:
         """
-        Upload model checkpoint or tokenizer files to the Hub while
-        synchronizing a local clone of the repo in `repo_path_or_name`.
+        Upload model checkpoint to the Hub.
 
         Parameters:
-            repo_path_or_name (`str`, *optional*):
-                Can either be a repository name for your model or tokenizer in
-                the Hub or a path to a local folder (in which case the
-                repository will have the name of that local folder). If not
-                specified, will default to the name given by `repo_url` and a
-                local directory with that name will be created.
-            repo_url (`str`, *optional*):
-                Specify this in case you want to push to an existing repository
-                in the hub. If unspecified, a new repository will be created in
-                your namespace (unless you specify an `organization`) with
-                `repo_name`.
+            repo_id (`str`, *optional*):
+                Repository name to which push
             commit_message (`str`, *optional*):
-                Message to commit while pushing. Will default to `"add config"`,
-                `"add tokenizer"` or `"add model"` depending on the type of the
-                class.
-            organization (`str`, *optional*):
-                Organization in which you want to push your model or tokenizer
-                (you must be a member of this organization).
+                Message to commit while pushing.
             private (`bool`, *optional*):
                 Whether the repository created should be private.
             api_endpoint (`str`, *optional*):
                 The API endpoint to use when pushing the model to the hub.
-            use_auth_token (`bool` or `str`, *optional*):
+            token (`str`, *optional*):
                 The token to use as HTTP bearer authorization for remote files.
-                If `True`, will use the token generated when running
-                `transformers-cli login` (stored in `~/.huggingface`). Will
-                default to `True` if `repo_url` is not specified.
-            git_user (`str`, *optional*):
-                will override the `git config user.name` for committing and
-                pushing files to the hub.
-            git_email (`str`, *optional*):
-                will override the `git config user.email` for committing and
-                pushing files to the hub.
+                If not set, will use the token set when logging in with
+                `transformers-cli login` (stored in `~/.huggingface`).
+            branch (Optional :obj:`str`):
+                The git branch on which to push the dataset. This defaults to
+                the default branch as specified in your repository, which
+                defaults to `"main"`.
```

Comment:
In regards to earlier comment about warning user about files larger than 30GB, maybe that can be put here more explicitly? Not sure which place is better (or maybe both places?)

Reply:
Will be discussed in a separate issue https://github.com/huggingface/huggingface_hub/issues/995.